ci: pick up cilium-cli v0.11.9 for master/v1.11 workflows

tklauser opened the pull request:
The --base-version flag is no longer necessary and has been deprecated,
see https://github.com/cilium/cilium-cli/pull/891 for details.

Release notes: https://github.com/cilium/cilium-cli/releases/tag/v0.11.9

sayboras reviewed (approved):
:shipit:

pchaigno commented on .github/workflows/conformance-eks-v1.11.yaml:
```diff
@@ -259,6 +256,8 @@ jobs:
         run: |
           cilium hubble enable ${{ steps.vars.outputs.hubble_enable_defaults }}
+          # NB: necessary to work against occassional flakes due to https://github.com/cilium/cilium-cli/issues/918
+          cilium status --wait
```
Nit: That would be best placed in a separate commit so we can just revert when the time comes (my understanding is that it's just a workaround).

tklauser replied:
Good point, will move to a separate commit.

pchaigno commented on .github/workflows/conformance-aws-cni-v1.10.yaml:
```diff
@@ -66,7 +66,7 @@ env:
   cilium_cli_version: v0.10.7
   check_url: https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}
-  eksctl_version: v0.94.0
+  eksctl_version: v0.101.0
```
Do we also need to update the docs?

tklauser replied:
We currently don't seem to suggest a particular `eksctl` version in the docs. Not sure whether we should?

pchaigno replied:
I'm not sure we had a strict requirement before. It seems like now we do.

tklauser replied:
> Do we also need to update the docs?

Addressed in #20329

pchaigno commented on .github/workflows/conformance-aws-cni.yaml:
```diff
@@ -250,7 +250,7 @@ jobs:
       - name: Update AWS VPC CNI plugin
         run: |
-          kubectl apply -f https://raw.githubusercontent.com/aws/amazon-vpc-cni-k8s/v1.7.10/config/v1.7/aws-k8s-cni.yaml
+          kubectl apply -f https://raw.githubusercontent.com/aws/amazon-vpc-cni-k8s/release-1.11/config/master/aws-k8s-cni.yaml
```
Isn't this happening because of the previous commit updating eksctl?

tklauser replied:
Likely, yes. I'll squash the two commits.

tklauser committed:
ci: pick up cilium-cli v0.11.9 for master/v1.11 workflows

The --base-version flag is no longer necessary and has been deprecated,
see https://github.com/cilium/cilium-cli/pull/891 for details.

Release notes: https://github.com/cilium/cilium-cli/releases/tag/v0.11.9

Signed-off-by: Tobias Klauser <tobias@cilium.io>

tklauser committed:
ci: explicitly wait for cilium status after enabling Hubble relay

With cilium-cli v0.11.9 on EKS we occasionally see flakes such as
https://github.com/cilium/cilium-cli/issues/918 when port forwarding
after enabling Hubble Relay. Adding an explicit `cilium status --wait`
works around these.

Add a comment so we don't forget to remove the workaround once the
underlying issue is resolved.

Signed-off-by: Tobias Klauser <tobias@cilium.io>

tklauser committed:
ci: update eksctl to v0.101.0 and AWS VPC CNI plugin to v1.11

This should resolve the following error we're seeing when installing
Cilium using cilium-cli built against k8s v1.24 libraries:

    unable to create Kubernetes client: exec plugin: invalid apiVersion "client.authentication.k8s.io/v1alpha1"

The eksctl in turn requires a newer version of the AWS VPC CNI plugin,
so update it to the latest v1.11.

Signed-off-by: Tobias Klauser <tobias@cilium.io>
